agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/event/EventScreen.kt

```diff
@@ -1,50 +1,110 @@
 package com.nibalk.tasky.agenda.presentation.event
 
+import androidx.compose.foundation.layout.WindowInsets
 import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.ime
+import androidx.compose.material3.HorizontalDivider
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
 import com.nibalk.tasky.agenda.presentation.R
 import com.nibalk.tasky.agenda.presentation.components.AgendaDetailHeader
+import com.nibalk.tasky.agenda.presentation.components.AgendaFooter
 import com.nibalk.tasky.agenda.presentation.model.AgendaArgs
 import com.nibalk.tasky.agenda.presentation.model.AgendaType
-import com.nibalk.tasky.core.presentation.components.TaskyBackground
-import java.time.LocalDate
+import com.nibalk.tasky.core.presentation.components.TaskyEditableDateTimeRow
+import com.nibalk.tasky.core.presentation.components.TaskyEditableTextRow
+import com.nibalk.tasky.core.presentation.components.TaskyEditableTextRowType
+import com.nibalk.tasky.core.presentation.components.TaskyScrollableBackground
+import com.nibalk.tasky.core.presentation.themes.TaskyLightBlue
+import com.nibalk.tasky.core.presentation.themes.TaskyTheme
+import com.nibalk.tasky.test.mock.AgendaSampleData
+import org.koin.androidx.compose.koinViewModel
+import org.koin.core.parameter.parametersOf
+import java.time.LocalTime
 
 @Composable
 fun EventScreenRoot(
-    agendaArgs: AgendaArgs
+    agendaArgs: AgendaArgs,
+    viewModel: EventViewModel = koinViewModel { parametersOf(agendaArgs) },
 ) {
     EventScreen(
-        isEditable = agendaArgs.isEditable,
-        agendaId = agendaArgs.agendaId,
-        selectedDate = agendaArgs.selectedDate ?: LocalDate.now()
+        state = viewModel.state,
+        onAction = { action ->
+            viewModel.onAction(action)
+        },
     )
 }
 
 @Composable
 fun EventScreen(
-    isEditable: Boolean,
-    agendaId: String?,
-    selectedDate: LocalDate,
+    state: EventState,
+    onAction: (EventAction) -> Unit,
 ) {
```

Comment:
One of the tougher decisions to make for this project is whether to use one screen per agenda item type or one screen for all agenda items. 2 years ago, I was also team single screen per type, but my mind has changed 😄 

My concern was that having a single screen makes it harder to read and update, but it actually got a lot easier:
If you look at the UI hierarchy of the items, reminder and task screens are almost identical while the event screen also has the same elements with a few extra composables. 

The complete root screen structure is no more than this:
```
HeaderSection()
TitleSection()
DescriptionSection()
if(item is Event) {
    PhotoSection()
}
StartDateSection()
if(item is Event) {
    EndDateSection()
}
ReminderSection()
if(item is Event) {
    AttendeeSection()
}
DeleteButton()
```
As long as you extract the UI into single components, it remains very readable. Same goes for the underlying ViewModel, you will save so much code by just having one ViewModel, since the logic of what all items have in common only needs to be written once. And if you'd extend that screen with another type of agenda item, that would also be easier and wouldn't require you to duplicate most of this to another screen.